### ISSUE TITLE
Create a file upload plugin

### ISSUE THREAD

**Issue by tokuhirom** [kwebio/kweb-core#392]
I want to use `callbackWs` in my code, but I cant determine it's a public API or not. It's not listed on `docs.kweb.io` site.

**Comment by tokuhirom** [comment]
I want to write a code like this:

```js
function makeImageUploadable(textAreaElementId, callbackId) {
    document.getElementById(textAreaElementId).addEventListener("paste", function (e) {
        const el = document.activeElement;
        for (let i = 0; i < e.clipboardData.items.length; ++i) {
            const item = e.clipboardData.items[i];
            console.log("onPaste: kind:", item.kind, "type:", item.type);

            if (item.kind === "text") {
                const [start, end] = [el.selectionStart, el.selectionEnd];
                el.setRangeText(item.getData("text"), start, end);
            } else if (item.kind === "file" && item.type.match(/^image\//)) {
                const reader = new FileReader();
                reader.onload = function (e) {
                    console.log("Uploading image");
                    callbackWs(callbackId, e.target.result)
                };
                reader.readAsDataURL(item.getAsFile());
            }
        }
    });
}
```

**Comment by sanity** [comment]
`callbackWs` isn't intended for this kind of use as it may change.

I think the best approach is to create a [KwebPlugin](https://docs.kweb.io/api/kweb-core/kweb.plugins/-kweb-plugin/index.html) and override its [appServerConfigurator](https://docs.kweb.io/api/kweb-core/kweb.plugins/-kweb-plugin/app-server-configurator.html) function to create a route that handles the [file upload](https://github.com/ktorio/ktor-documentation/tree/2.1.3/codeSnippets/snippets/upload-file).

It would also require a browser-side [function](https://docs.kweb.io/api/kweb-core/kweb/-web-browser/call-js-function.html) to handle sending the file.

If this plugin is created in a general enough way I'd be happy to add it to Kweb for others to use.

Does that make sense?

**Comment by tokuhirom** [comment]
Ah, implementing KwebPlugin is a good idea.

I tried to use this method for my product.
- https://github.com/tokuhirom/blog3/blob/main/src/main/resources/static/js/admin.js
- https://github.com/tokuhirom/blog3/blob/main/src/main/kotlin/blog3/admin/plugin/FileUploadPlugin.kt

And it works well. Thanks.

> If this plugin is created in a general enough way and you create a pull request, I'd be happy to add it to Kweb for others to use. If you're not sure if you can do this I can add it to my todo list but it may take a few days before I can get to it.

...But, making plugin for general purpose uploading process is bit hard.
Because, if my understanding is correct,  kweb doesn't want to expose ktor's classes to kweb developers. As a result, if i make a plugin, i need to wrap a lot  from `ApplicationCall`. And the uploading process in front end is very simple. [just call window.fetch](https://github.com/tokuhirom/blog3/blob/main/src/main/resources/static/js/admin.js#L12-L16). Most of developers doesn't need the wrapper function :P

In summary, in my opinion, it would be more beneficial if this was provided in a form like a How-to document rather than being provided as a Plugin.

**Comment by sanity** [comment]
Glad that worked for you. 

I don't have a problem exposing Ktor classes to Kweb developers, there are already some places it's exposed (like the plugin mechanism).

I'll give this some more thought, I have also been considering a "recipes" section of the user manual for this type of thing.
